resources/design-document.md

```diff
@@ -0,0 +1,335 @@
+String countUnit;
+Double grossPayment;
+Double taxRate;
+String projectType;
+String dueDate;
+String startDate;
+String endDate;
+Boolean openCase;
+Boolean casePaid;
+Boolean rushJob;
+Map<String, Map<String, String>> progressLog;
+Double totalWorkingHours;
+Double wordsPerHour;
+
+// TranslationClientModel
+
+String translationClientId;
+String translationClientName;
+String translationClientType;
+Map<String, Map<String, String>> translationCaseHistory;
+Map<String, Map<String, String>> paymentHistory;
+Map<String, String> outstandingPayments;
+```
+
+### Stretch Goal
+
+```
+// EstimateModel
+
+String estimateId;
+String translationClientId;
+String translationClientName;
+String translationClientType;
+String sourceTextTitle;
+String sourceTextAuthor;
+Double estimatedRate;
+String estimatedRateUnit;
+Integer count;
+String countUnit;
+Double estimatedGrossPayment;
+Double taxRate;
+String projectType;
+String dueDate;
+Double totalWorkingHoursEstimate;
+Double wordsPerHourEstimate;
+```
+
+## 6.2. Get Translation Case Endpoint
+
+* Accepts `GET` requests to `/cases/:id`
+* Accepts a translation case ID and returns the corresponding `TranslationCaseModel`.
+    * If the given translation case ID is not found, will throw a
+      `TranslationCaseNotFoundException`
+
+<img src="images/GetTranslationCase-Sequence.png" width="100%" alt="Get Translation Case Endpoint Sequence Diagram"/>
+
+## 6.3 Create Translation Case Endpoint
+
+* Accepts `POST` requests to `/cases`
+* Accepts data to create a new `TranslationCase` with a provided case nickname, a given translation case ID, with all 
+  other values being optional. 
+* Returns the new `TranslationCase`, including a unique translation case ID assigned by Translation Tracker.
+  * If the case-nickname translation-client-id pair is identical to an already-existing pair, will throw a 
+    `DuplicateCaseNicknameException`.
+
+<img src="images/CreateTranslationCase-Sequence.png" width="100%" alt="Create Translation Case Endpoint Sequence Diagram"/>
+
+## 6.4 Update Translation Case Endpoint
+
+* Accepts `PUT` requests to `/cases/:id`
+* Accepts data to update a `TranslationCase` including a translation case ID, and the update values. Returns the updated
+  `TranslationCase`.
+    * If the given translation case ID is not found, will throw a `TranslationCaseNotFoundException`
+
+<img src="images/UpdateTranslationCase-Sequence.png" width="100%" alt="Update Translation Case Endpoint Sequence Diagram"/>
+
+## 6.5 Archive Translation Case Endpoint
+
+* Accepts `DELETE` requests to `/cases/:id`
+* Accepts a translation case ID and archives the specified `TranslationCase`.
+  * If the translation case ID is not found, will throw a `TranslationCaseNotFoundException`
+
+<img src="images/ArchiveTranslationCase-Sequence.png" width="100%" alt="Archive Translation Case Endpoint Sequence Diagram"/>
+
+## 6.6 Get Translation Client Endpoint
+
+* Accepts `GET` requests to `/translationclients/:id`
```

Comment:
Is there particular significance to using `translationclients` for clients, but just `cases` for cases?

Reply:
No significance, it's now been updated to be consistent (translationclients / translationcases)